wallet/src/Pos/Wallet/Web/Pending/Submission.hs

```diff
@@ -166,9 +174,73 @@ submitAndSavePtx submitTx PtxSubmissionHandlers{..} ptx@PendingTx{..} = do
         -- while transaction creation failed, due to protocol error or bug,
         -- then we better not remove this pending transaction
         void $ usingPtxCoords removeOnlyCreatingPtx ptx
+
     badInitPtxCondition = InternalError "Expected PtxCreating as initial pending condition"
 
     reportSubmitted ack =
         logDebug $
         sformat ("submitAndSavePtx: transaction submitted with confirmation?: "
                 %build) ack
+
+
+-- As simple and as clear as I could.
+submitAndSavePtxMocked
+    :: TxSubmissionMode ctx m
+    => PendingTx
+    -> Timestamp
+    -> ((TxId, TxAux) -> m TxSubmissionResult)
+    -> m TxSubmissionResult
+submitAndSavePtxMocked ptx@PendingTx{..} now mSaveTx =
+
+    case _ptxCond of
+
+      PtxApplying poolInfo -> do
+          let Just creationTime = poolInfo ^. thTimestamp
+          if diffTimestamp now creationTime > hour 1
+              then pure $ TxTimeoutWhenApplying poolInfo ptx
+              else pure $ TxStillApplying
+
+      _ -> do
```

Comment:
Erm, kind of hard to handle all the cases since there literally are two:
- `PtxApplying`
- the rest

I can list them all out, but I'm not sure how useful is that since I'll copy the last part on all the other cases. I fear we might awake the HLint monster, "reduce duplication". If you think it's more readable, I can add it.